whackaworm.js

```diff
@@ -1,62 +1,58 @@
-var imgArr;
-var hole1;
-var hole2;
-var hole3;
-var hole4;
-var hole5;
-var hole6;
-var hole7;
-var hole8;
-var hole9;
-var worm;
-var score;
-var randomNumber;
-var previousRandomNum;
+var holes, worm, hole, score, randomNum, previousRandomNum, timeout, functionName;
 
-function init()
-{
-	score = 0;
-	previousRandomNum = 0;
-	randomNumber = 0;
+function init(){
+	initializeAssets();
 	cacheImages();
 	
-	addEvent(hole1, "click", randomize, false);
-	addEvent(hole2, "click", randomize, false);
-	addEvent(hole3, "click", randomize, false);
-	addEvent(hole4, "click", randomize, false);
-	addEvent(hole5, "click", randomize, false);
-	addEvent(hole6, "click", randomize, false);
-	addEvent(hole7, "click", randomize, false);
-	addEvent(hole8, "click", randomize, false);
-	addEvent(hole9, "click", randomize, false);
+	for(i=0; i < holes.length;i ++){
+		addEvent(holes[i], functionName, randomize, false);
+	}
 
 	play();
-	setTimeout("endGame()",60000);
+	setTimeout("endGame()", timeout);
+}
 
+function initializeAssets(){
+	timeout = 60000;
+	functionName = "click";
+	score = 0;
+	previousRandomNum = 0;
+	randomNum = 0;
+	hole = "hole.png";
+	worm = "wormHoleImage.png";
 }
 
-function generateNum()
-{
-	randomNumber = Math.floor(Math.random()*10);
-	while(previousRandomNum == randomNumber || randomNumber == 9){
-		randomNumber = Math.floor(Math.random()*10);
+function cacheImages(){
+    //Cache all the image tags
+    hole1 = document.getElementById("hole1");
```

Comment:
named this way because of the Id's in the HTML.. need to research if this can be done differently. cannot update atm.